src/main/java/emu/grasscutter/game/quest/handlers/QuestBaseHandler.java

```diff
@@ -1,10 +1,23 @@
 package emu.grasscutter.game.quest.handlers;
 
 import emu.grasscutter.data.excels.QuestData.QuestCondition;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.game.quest.GameMainQuest;
 import emu.grasscutter.game.quest.GameQuest;
+import jdk.jshell.spi.ExecutionControl;
 
 public abstract class QuestBaseHandler {
 
-	public abstract boolean execute(GameQuest quest, QuestCondition condition, String paramStr, int... params);
+	public  boolean execute(GameQuest quest, QuestCondition condition, String paramStr, int... params) throws ExecutionControl.NotImplementedException {
```

Comment:
why it divided into 3 methods here? you can access the main quest by `quest.getMainQuest()` and the player by `quest.getMainQuest().getOwner()`

Reply:
WIth how GameMainQuest::evaluateAccept/Fail/FinishCond is called, unless subQuest is in params, we can't determine which subQuest. Also, in many QuestTriggers, subQuest is not relevant. Finally, I think this improves code readability.

Reply:
i think there is no need to determine which subquest here, because a loop will check all of the subquests in main quests and auto accept the quests which meet the condition. And it cause hard to maintain by repeating too much code in `ServerQuestHandler` and an unnecessarily `switch` in `evaluateAcceptCond` etc. 
ideally, we can just add a Handler to support the new trigger, no need to add a `case`

Reply:
ok I see. I'll revert this, and move the debug messages to the appropriate handlers

Reply:
I agree for fail and finish, but for evaluateAcceptCond, quest will always be `null`

Reply:
for accept, it should check the quest metadata; for fail or finish, it should check the quest which player holds it

Reply:
we'd still need to have a different QuestBaseHandler::execute for accepting quests though, so `ServerQuestHandler` would still be messy. I think initializing all subQuests to `QUEST_STATE_UNSTARTED` is cleaner, and it's probably what Certain Anime Game Company is doing. We could remove all the subQuests from the MainQuest when `PARENT_QUEST_STATE_FINISHED` or `PARENT_QUEST_STATE_FAILED` or `PARENT_QUEST_STATE_CANCELED` from the database to save space afterwards